lib/dependencies-parser.ts

```diff
@@ -0,0 +1,179 @@
+import {startsWith} from 'lodash';
+import {Line, parse as parseLines} from './line-parser';
+
+const COMMENTS = ['#', '//'];
+const GROUP = 'group';
+const SOURCE = 'source';
+// https://fsprojects.github.io/Paket/dependencies-file.html#Sources
+const GITHUB = 'github';
+const NUGET = 'nuget';
+const CLITOOL = 'clitool';
+const GIT = 'git';
+const GIST = 'gist';
+const HTTP = 'http';
+
+interface Options {
+  [name: string]: string;
+}
+
+interface Source {
+  url: string;
+  options: Options;
+}
+
+interface DependencyGroup {
+  name: string | null;
+  options: Options;
+  sources: Source[];
+  dependencies: Dependency[];
+}
+
+interface Dependency {
+  source: string;
+}
+
+interface GithubDependency extends Dependency {
+  source: 'github';
+  repo: string;
+  file: string;
+  version: string;
+  token: string;
+}
+
+interface NugetDependency extends Dependency {
+  source: 'nuget';
+  name: string;
+  versionRange: string;
+  options: Options;
+}
+
+export interface PaketDependencies extends Array<DependencyGroup> {
+}
+
+function parseNuget(line: string): NugetDependency {
+  const result: NugetDependency = {
+    name: '',
+    options: {},
+    source: 'nuget',
+    versionRange: '',
+  };
+
+  // Get position of first option.
+  const firstOptionMatch = /\s*[^\s]+\s*:/.exec(line);
+  let dependencyString = line;
+  // let optionsString = '';
+
+  if (firstOptionMatch) {
+    // Split line by position of first option.
+    dependencyString = line.substr(0, firstOptionMatch.index);
+    // optionsString = line.substr(firstOptionMatch.index).trim();
+  }
+
+  dependencyString = dependencyString.replace(NUGET, '').trim();
+
+  // Split by space between words. First chunk will be name of the dependency and all rest - version.
+  const dependencyStringParts = dependencyString.split(/\s+/);
+
+  result.name = dependencyStringParts[0];
+  result.versionRange = dependencyStringParts.splice(1).join(' ');
+
+  // TODO: parse options
+
+  return result;
+}
+
+// https://fsprojects.github.io/Paket/github-dependencies.html
+function parseGithub(line: string): GithubDependency {
+  const re = /"[^"]*"|\S+/g;
+  const parts = line.match(re).splice(1);
+  const [repo, version] = parts[0].split(':');
+
+  return {
```

Comment:
Should we do some validation in case the file we get is not valid according to what we know?

We could just log it in debug mode for now if it's irrelevant to getting the dependencies themselves.

Reply:
I would ratter ignore this for now and fail with default `out of bounds` exception. We need to replace line parser with proper tokenizer anyway, let's do it there?